src/core/Akka.Streams/Dsl/Source.cs

```diff
@@ -550,7 +550,7 @@ public static class Source
         public static Source<T, NotUsed> Repeat<T>(T element)
         {
             var next = new Tuple<T, T>(element, element);
-            return Unfold(element, _ => next).WithAttributes(DefaultAttributes.Repeat);
+            return Unfold(element, _ => new Option<Tuple<T, T>>(next)).WithAttributes(DefaultAttributes.Repeat);
```

Comment:
`UnfoldInfinite` could be used here too, by the way. Maybe even more convenient...